#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

# Mingw gcc-9 doesn't provide fortified functions (memmove_chk, memcopy_chk)
# When defining _FORTIFY_SOURCE to a value greater than 0,
# these functions have to be provided by libssp (-fstack-protector or -lssp).
# Adding -lssp currentyly only creates a static cairo library, so until this
# is cleared, disable fortification and use gcc-8 for release versions.

diff -ur cairo-1.17.4-org/build/configure.ac.warnings cairo-1.17.4/build/configure.ac.warnings
--- cairo-1.17.4-org/build/configure.ac.warnings	2010-12-25 15:21:33.000000000 +0100
+++ cairo-1.17.4/build/configure.ac.warnings	2021-03-02 13:45:49.950952181 +0100
@@ -39,7 +39,7 @@
 MAYBE_WARN="$MAYBE_WARN -fno-strict-aliasing -fno-common"
 
 dnl Also to turn various gcc/glibc-specific preprocessor checks
-MAYBE_WARN="$MAYBE_WARN -Wp,-D_FORTIFY_SOURCE=2"
+MAYBE_WARN="$MAYBE_WARN -Wp"
 
 # invalidate cached value if MAYBE_WARN has changed
 if test "x$cairo_cv_warn_maybe" != "x$MAYBE_WARN"; then
